OPHD/States/MapViewState.cpp

```diff
@@ -287,6 +287,7 @@ void MapViewState::focusOnStructure(Structure* structure)
 {
 	if (!structure) { return; }
 	mMapView->centerOn(NAS2D::Utility<StructureManager>::get().tileFromStructure(structure).xyz());
+	populateStructureMenu();
```

Comment:
Potentially you could make use of `void MapViewState::onNotificationWindowTakeMeThere(const MapCoordinate& position)` to avoid duplication:
```cpp
onNotificationWindowTakeMeThere(NAS2D::Utility<StructureManager>::get().tileFromStructure(structure).xyz());
```

Though I would want to rename it to `onTakeMeThere` to better represent the more general usage.

Reply:
I was thinking about the best way to do that. I'd want to rename that function as you suggested to `onTakeMeThere` and just call it from the `focusOnStructure()` function. I'll open an issue for it.